dotnet/src/webdriver/CookieJar.cs

```diff
@@ -23,104 +23,61 @@
+            if (cookie is null)
+            {
+                throw new ArgumentNullException(nameof(cookie));
+            }
+
+            driver.InternalExecute(DriverCommand.AddCookie, new() { { "cookie", cookie } });
         }
 
-        /// <summary>
-        /// Delete the cookie by passing in the name of the cookie
-        /// </summary>
-        /// <param name="name">The name of the cookie that is in the browser</param>
         public void DeleteCookieNamed(string name)
         {
-            Dictionary<string, object> parameters = new Dictionary<string, object>();
-            parameters.Add("name", name);
-            this.driver.InternalExecute(DriverCommand.DeleteCookie, parameters);
+            if (name is null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            driver.InternalExecute(DriverCommand.DeleteCookie, new() { { "name", name } });
         }
 
-        /// <summary>
-        /// Delete a cookie in the browser by passing in a copy of a cookie
-        /// </summary>
-        /// <param name="cookie">An object that represents a copy of the cookie that needs to be deleted</param>
         public void DeleteCookie(Cookie cookie)
         {
-            if (cookie != null)
+            if (cookie is null)
             {
-                this.DeleteCookieNamed(cookie.Name);
+                throw new ArgumentNullException(nameof(cookie));
             }
+
+            DeleteCookieNamed(cookie.Name);
         }
 
-        /// <summary>
-        /// Delete All Cookies that are present in the browser
-        /// </summary>
         public void DeleteAllCookies()
         {
-            this.driver.InternalExecute(DriverCommand.DeleteAllCookies, null);
+            driver.InternalExecute(DriverCommand.DeleteAllCookies, null);
         }
 
-        /// <summary>
-        /// Method for returning a getting a cookie by name
-        /// </summary>
-        /// <param name="name">name of the cookie that needs to be returned</param>
-        /// <returns>A Cookie from the name</returns>
         public Cookie GetCookieNamed(string name)
         {
-            Cookie cookieToReturn = null;
-            if (name != null)
+            if (name is null)
             {
-                ReadOnlyCollection<Cookie> allCookies = this.AllCookies;
-                foreach (Cookie currentCookie in allCookies)
-                {
-                    if (name.Equals(currentCookie.Name))
-                    {
-                        cookieToReturn = currentCookie;
-                        break;
-                    }
-                }
+                throw new ArgumentNullException(nameof(name));
             }
 
-            return cookieToReturn;
+            var rawCookie = driver.InternalExecute(DriverCommand.GetCookie, new() { { "name", name } }).Value;
+
+            return Cookie.FromDictionary(rawCookie as Dictionary<string, object>);
```

Comment:
This will fight nullability once that's implemented on `Cookie`. Could this be a cast instead?

```suggestion
            return Cookie.FromDictionary((Dictionary<string, object>)rawCookie);
```

Reply:
Done.